pkg/operator/controller/gatewayapi/controller_test.go

```diff
@@ -88,6 +122,85 @@ func Test_Reconcile(t *testing.T) {
 			expectDelete:    []client.Object{},
 			expectStartCtrl: false,
 		},
+		{
+			name:                        "unmanaged gateway API CRDs created",
+			gatewayAPIEnabled:           true,
+			gatewayAPIControllerEnabled: true,
+			existingObjects: []runtime.Object{
+				co("ingress"),
+				crd("listenersets.gateway.networking.x-k8s.io"),
+			},
+			existingStatusSubresource: []client.Object{
+				co("ingress"),
+			},
+			expectCreate: []client.Object{
+				crd("gatewayclasses.gateway.networking.k8s.io"),
+				crd("gateways.gateway.networking.k8s.io"),
+				crd("grpcroutes.gateway.networking.k8s.io"),
+				crd("httproutes.gateway.networking.k8s.io"),
+				crd("referencegrants.gateway.networking.k8s.io"),
+				clusterRole("system:openshift:gateway-api:aggregate-to-admin"),
+				clusterRole("system:openshift:gateway-api:aggregate-to-view"),
+			},
+			expectUpdate: []client.Object{},
+			expectDelete: []client.Object{},
+			expectStatusUpdate: []client.Object{
+				coWithExtension("ingress", `{"unmanagedGatewayAPICRDNames":"listenersets.gateway.networking.x-k8s.io"}`),
+			},
+			expectStartCtrl: true,
+		},
+		{
+			name:                        "unmanaged gateway API CRDs removed",
```

Comment:
I'm not sure what's happening here.  Is this simulating the removal of unmanaged CRDs?

Reply:
Yes. We have the ingress cluster operator with `unmanagedGatewayAPICRDNames` extension but no  unmanaged CRD exists (== added to `existingObjects`). So, we have to reset `unmanagedGatewayAPICRDNames` extension.